File: src/main/java/com/github/zyypj/tadeuBooter/minecraft/serialization/LocationSerializer.java
package com.github.zyypj.tadeuBooter.minecraft.serialization;

import org.bukkit.Bukkit;
import org.bukkit.Location;

public class LocationSerializer {

    private static final String SPACE_PLACEHOLDER = "-";

    public static String serialize(Location location) {
        String worldName = location.getWorld().getName().replace(" ", SPACE_PLACEHOLDER);
        return worldName + " " + location.getX() + " " + location.getY() + " " + location.getZ() + " " + location.getYaw() + " " + location.getPitch();
    }

    public static Location deserialize(String string) {
        String[] split = string.split(" ");
        String worldName = split[0].replace(SPACE_PLACEHOLDER, " ");
        return new Location(
                Bukkit.getWorld(worldName),
                Double.parseDouble(split[1]),
                Double.parseDouble(split[2]),
                Double.parseDouble(split[3]),
                Float.parseFloat(split[4]),
                Float.parseFloat(split[5])
        );
    }
}

File: spigot/src/main/java/me/zyypj/booter/minecraft/spigot/location/util/serialization/LocationSerializer.kt
package me.zyypj.booter.minecraft.spigot.location.util.serialization

import org.bukkit.Bukkit
import org.bukkit.Location

object LocationSerializer {

    private val SPACE_PLACEHOLDER = "-"

    /**
     * Converte uma Location em String no formato: world x y z yaw pitch
     */
    @JvmStatic
    fun serialize(location: Location): String {
        val worldName = location.world!!.name.replace(" ", SPACE_PLACEHOLDER)
        return listOf(
            worldName,
            location.x.toString(),
            location.y.toString(),
            location.z.toString(),
            location.yaw.toString(),
            location.pitch.toString()
        ).joinToString(" ")
    }

    /**
     * Reconstrói uma Location a partir da String serializada.
     */
    @JvmStatic
    fun deserialize(string: String): Location {
        val parts = string.split(" ")
        val worldName = parts[0].replace(SPACE_PLACEHOLDER, " ")
        val world = Bukkit.getWorld(worldName)
        return Location(
            world,
            parts[1].toDouble(),
            parts[2].toDouble(),
            parts[3].toDouble(),
            parts[4].toFloat(),
            parts[5].toFloat()
        )
    }
}